build_tools/ci/run_matmul_test.sh

```diff
@@ -0,0 +1,153 @@
+#!/bin/bash
+# Copyright 2024 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# Manual IREE matmul test script.
+#
+# This pulls code from
+#   * https://github.com/openxla/iree/blob/main/build_tools/cmake/iree_e2e_matmul_test.cmake
+#   * https://github.com/openxla/iree/blob/main/tests/e2e/matmul/CMakeLists.txt
+#
+# Usage:
+#   1. Build IREE (or install packages). You'll need `iree-compile` to include
+#      support for the compiler target backend you wish to test and
+#      `iree-e2e-matmul-test` to include support for the runtime HAL
+#      driver/device you wish to test.
+#   2. Update the paths in this script or specify them via environment variables
+#   3. Run: `./run_matmul_tests.sh <peano_install_path> <mlir_aie_install_path> <vitis_path>`
+
+set -euo pipefail
+
+THIS_DIR="$(cd $(dirname $0) && pwd)"
+ROOT_DIR="$(cd $THIS_DIR/../.. && pwd)"
+
+OUTPUT_DIR="${IREE_MATMUL_BUILD_DIR:-build-matmul}"
+GENERATOR="${ROOT_DIR}/tests/matmul/generate_e2e_matmul_tests.py"
+IREE_PYTHON3_EXECUTABLE="${IREE_PYTHON3_EXECUTABLE:-python3}"
+
+IREE_INSTALL_BIN="${IREE_INSTALL_BIN:-${ROOT_DIR}/../iree-build-debug/tools/}"
+IREE_COMPILE_EXE="${IREE_INSTALL_BIN}/iree-compile"
+TEST_RUNNER="${IREE_INSTALL_BIN}/iree-e2e-matmul-test"
```

Comment:
Does the iree-amd-aie CI have these files available somewhere already? I'm wondering how easily this can be plugged into a GitHub Actions workflow. Did you want to set that up in this PR also?

Reply:
I think these binary's can be found however, the dependencies are not installed in any CI bots and any CI bots dont have an AIE so I think this is next task for @saienduri